secure/storage/src/tests/vault.rs

```diff
@@ -7,13 +7,37 @@ use crate::{Capability, Error, Identity, Permission, Policy, Storage, Value};
 /// A test for verifying VaultStorage properly implements the LibraSecureStorage API. This test
 /// depends on running Vault, which can be done by using the provided docker run script in
 /// `docker/vault/run.sh`
+/// Note: because we only run a single vault backend for testing, we execute all tests sequentially
+/// using the method below.
+///
+/// TODO(joshlind): see how we can refactor this to run each test individually. This would better
+/// separate the testing of individual behaviours and help make failures easier to reason about.
+/// It would also be good to avoid having to call reset_vault_storage between tests..
 #[ignore]
 #[test]
-fn vault() {
+fn all_tests() {
+    let mut storage = Box::new(setup_vault());
+
+    // Test key/value related operations
+    suite::run_test_suite(storage.as_mut(), "VaultStorage");
+    reset_vault_storage(storage.as_mut());
+
+    // Test cryptographic key related operations
+    suite::create_get_and_test_key_pair(storage.as_mut());
+    reset_vault_storage(storage.as_mut());
```

Comment:
perhaps the alternative is that we introduce a reset on all storages and we still call suite::run_all_tests, see safety rules: https://github.com/libra/libra/blob/master/consensus/safety-rules/src/tests/suite.rs#L39

Reply:
This is something that I already tackle in my refactor PR (which is ready, but hasn't been pushed yet as it's blocked on this). If possible, could we defer dealing with this to the refactor? I think the refactor does a pretty good job of address this already.